apps/web/pages/team/[slug].tsx

```diff
@@ -249,29 +245,33 @@ function TeamPage({ team, isUnpublished, markdownStrippedBio, isValidOrgDomain }
 export const getServerSideProps = async (context: GetServerSidePropsContext) => {
   const ssr = await ssrInit(context);
   const slug = Array.isArray(context.query?.slug) ? context.query.slug.pop() : context.query.slug;
-  const { isValidOrgDomain } = orgDomainConfig(context.req.headers.host ?? "");
+  const { isValidOrgDomain, currentOrgDomain } = orgDomainConfig(
+    context.req.headers.host ?? "",
+    context.query.orgSlug
+  );
   const flags = await getFeatureFlagMap(prisma);
-
-  const team = await getTeamWithMembers(undefined, slug);
+  const team = await getTeamWithMembers(undefined, slug, undefined, currentOrgDomain);
   const metadata = teamMetadataSchema.parse(team?.metadata ?? {});
-
   // Taking care of sub-teams and orgs
   if (
-    (isValidOrgDomain && team?.parent && !!metadata?.isOrganization) ||
     (!isValidOrgDomain && team?.parent) ||
     (!isValidOrgDomain && !!metadata?.isOrganization) ||
     flags["organizations"] !== true
   ) {
     return { notFound: true } as const;
   }
 
-  if (!team) {
+  if (!team || !team.parent?.slug) {
```

Comment:
We also consider unpublished a team with no parent slug

Reply:
we are throwing 404 for normal teams now: 

<img width="921" alt="Screenshot 2023-07-28 at 16 27 57" src="https://github.com/calcom/cal.com/assets/30310907/52c02522-3cc6-47b9-83a6-9c95fce9dc64">

Is that what we want? 
```suggestion
  if (!team || (team.parent && !team.parent.slug) {
```

Reply:
Oh left an `orgDomains` util/test changes in the stash! Thanks, accepted your suggestion as well as it was needed.